### PR TITLE
Add connectAndSignIn to CBWProvider

### PR DESCRIPTION
**Pull request by bangtoven**

### CODE REVIEW

File: packages/wallet-sdk/src/provider/CoinbaseWalletProvider.ts [hunk 1]
```diff
@@ -596,6 +596,37 @@ export class CoinbaseWalletProvider
     return res.result;
   }
 
+  /**
+   * @beta This method is currently in beta. While it is available for use, please note that it is still under testing and may undergo significant changes.
```
Comment (bangtoven) [inline packages/wallet-sdk/src/provider/CoinbaseWalletProvider.ts:600]: https://tsdoc.org/pages/tags/beta/

File: packages/wallet-sdk/src/provider/CoinbaseWalletProvider.ts [hunk 2]
```diff
@@ -596,6 +597,76 @@ export class CoinbaseWalletProvider
     return res.result;
   }
 
+  /**
```
Comment (bangtoven) [inline packages/wallet-sdk/src/provider/CoinbaseWalletProvider.ts:600]: Screenshot 2023-06-30 at 3 57 08 PM

File: packages/wallet-sdk/src/provider/CoinbaseWalletProvider.ts [hunk 3]
```diff
@@ -596,6 +597,76 @@ export class CoinbaseWalletProvider
...
+   * @beta
+   * This method is currently in beta. While it is available for use, please note that it is still under testing and may undergo significant changes.
+   *
+   * @remarks
+   * IMPORTANT: Signature validation is not performed by this method. Users of this method are advised to perform their own signature validation.
+   * Common web3 frontend libraries such as ethers.js and viem provide the `verifyMessage` utility function that can be used for signature validation.
```
Comment (bangtoven) [inline packages/wallet-sdk/src/provider/CoinbaseWalletProvider.ts:606]: doesn't seem like there's an existing method to do this.
we have one named `ethereumAddressFromSignedMessage`, but it's defined on the relay and makes request via WL...
image

File: packages/wallet-sdk/src/provider/CoinbaseWalletProvider.ts [hunk 4]
```diff
@@ -596,6 +597,76 @@ export class CoinbaseWalletProvider
...
+    statement?: string;
+    resources?: string[];
+  }): Promise<{
+    accounts: AddressString[];
+    message: string;
+    signature: string;
+  }> {
+    // NOTE: It was intentionally built by following the pattern of the existing eth_requestAccounts method
+    // to maintain consistency and avoid introducing a new pattern.
+    // We acknowledge the need for a better design, and it is planned to address and improve it in a future refactor.
```
Comment (bangtoven) [inline packages/wallet-sdk/src/provider/CoinbaseWalletProvider.ts:631]: image

**Review by vishnumad — APPROVED** [review]

**Pull request merged**